File: tst/server/www/pl/ul.pl
#!/usr/bin/perl

use strict;
use warnings;
use CGI;


my $cgi = CGI->new;

my $g1 = $cgi->param('g1');
my $g2 = $cgi->param('g2');

my $p1 = $cgi->param('p1');
my $p2 = $cgi->param('p2');

my $f = $cgi->param('file');

print("Content-Type: text/plain\r\n\r\n");

print("Perl : UPLOAD!\n");

if ($f)
{
    print("\nFILE\n");
    print($f, "\n");

    my $fp = $cgi->upload('file');
    print($fp);
    open UPLOADFILE, ">./upload-pl-$f";
    binmode UPLOADFILE;
    while ( <$fp> ) { print UPLOADFILE; }
    close UPLOADFILE;
}

File: tst/server/www/py/ul.py
#!/usr/bin/python
from __future__ import print_function

import sys
import os

import cgi
import cgitb


print("Content-Type: text/plain", end="\r\n\r\n");

form = cgi.FieldStorage()

print("Python : UPLOAD!")

if "file" in form:
    file_item = form["file"]
    if file_item.filename:
        print("file name", file_item.filename)
        file_name = file_item.filename
        file_path = './upload-py-' + file_name
        with open(file_path, 'wb') as file:
            file.write(file_item.file.read())